pkg/manager/member/tidb_discovery_manager.go

```diff
@@ -78,11 +81,11 @@ func (m *realTidbDiscoveryManager) Reconcile(obj runtime.Object) error {
 			Verbs:         []string{"get"},
 		}
 	default:
-		panic(fmt.Sprintf("unsupported type %T for discovery", obj))
+		return controller.RequeueErrorf("unsupported type %T for discovery", obj)
```

Comment:
No need to requeue if the resource is unsupported, WDYT?

Reply:
fixed in https://github.com/pingcap/tidb-operator/pull/3914/commits/8ef239b152f0f867335cc1e6a750ce6362e756e3. and some of the other errors returned from `switch` not updated because it may be complicated and have already covered by these updated ' switch`es.